lfs.c

```diff
@@ -2387,9 +2400,10 @@ int lfs_file_opencfg(lfs_t *lfs, lfs_file_t *file,
                  ".buffer=%p, .attrs=%p, .attr_count=%"PRIu32"})",
             (void*)lfs, (void*)file, path, flags,
             (void*)cfg, cfg->buffer, (void*)cfg->attrs, cfg->attr_count);
+    LFS_ASSERT(is_node_not_in_list(lfs->mlist, (struct lfs_mlist*)file));
 
     // deorphan if we haven't yet, needed at most once after poweron
-    if ((flags & 3) != LFS_O_RDONLY) {
+    if ((flags & LFS_O_RDWR) != LFS_O_RDONLY) {
```

Comment:
👍

Reply:
I changed in all places now :smile:

Reply:
Ah I misread this. Well, I don't really have an opinion on this change. It is a curious mask.

Reply:
I think this is better than a magic num

Reply:
Can't argue with that